docs/sources/reference/components/loki/loki.secretfilter.md

```diff
@@ -0,0 +1,122 @@
+
+<span class="badge docs-labels__stage docs-labels__item">Experimental</span>
+
+# loki.secretfilter
+
+{{< docs/shared lookup="stability/experimental.md" source="alloy" version="<ALLOY_VERSION>" >}}
+
+`loki.secretfilter` receives log entries and redacts sensitive information, such as secrets, from them. The detection is based on regex patterns, defined in the gitleaks configuration file embedded within the component. It can also use a custom configuration file based on the gitleaks configuration file structure.
+
+Note that Personally Identifiable Information (PII) or undefined secret types could remain undetected.
+
+[Gitleaks configuration file]: https://github.com/gitleaks/gitleaks/blob/master/config/gitleaks.toml
+
+## Usage
+
+```alloy
+loki.secretfilter "<LABEL>" {
+    forward_to = <RECEIVER_LIST>
+}
+```
+
+## Arguments
+
+`loki.secretfilter` supports the following arguments:
+
+Name                     | Type                 | Description                                     | Default                          | Required
+-------------------------|----------------------|-------------------------------------------------|----------------------------------|---------
+`forward_to`             | `list(LogsReceiver)` | List of receivers to send log entries to.       |                                  | yes
+`gitleaks_config`        | `string`             | Path to the custom gitleaks.toml file.          | Embedded one                     | no
+`types`                  | `map(string)`        | Types of secret to look for.                    | All types                        | no
+`redact_with`            | `string`             | String to use to redact secrets.                | `<REDACTED-SECRET:$SECRET_NAME>` | no
+`exclude_generic`        | `bool`               | Exclude the generic API key rule.               | `false`                          | no
+`allowlist`              | `map(string)`        | List of regexes to allowlist matching secrets.  | `{}`                             | no
+`partial_mask`           | `number`             | Show the first N characters of the secret.      | `0`                              | no
+
+The `gitleaks_config` argument is the path to the custom `gitleaks.toml` file. If not provided, the embedded one is used.
+
+The `types` argument is a map of secret types to look for. The values are used as prefixes for the secret types in the gitleaks config. If not provided, all types are used.
+
+The `redact_with` argument is a string that can use variables such as `$SECRET_NAME` (replaced with the matching secret type) and `$SECRET_HASH`(replaced with the sha1 hash of the secret).
+
+The `exclude_generic` argument is a boolean that excludes the generic API key rule in the gitleaks config file if set to `true`.
+
+The `allowlist` argument is a map of regexes to allowlist matching secrets. If a secret matches any of the regexes, it will not be redacted. The allowlist in the gitleaks configuration is also applied.
```

Comment:
```suggestion
The `allowlist` argument is a map of regular expressions to allow matching secrets.
Aa secret will not be redacted if it matches any of the regular expressions. The allowlist in the Gitleaks configuration file is also applied.
```

Reply:
Thanks for all the fixes, @clayton-cornell!